### DISCUSSION TITLE
Build and make DAT Mapper available to existing Keycloak instance

### DISCUSSION THREAD

**Discussion by xnakos** [sovity/sovity-daps#12] [Q&A] [answered]
Hello, everyone. What is the recommended way to build and add the DAT Mapper to an existing Keycloak instance? That is, if one does not want to deploy the whole `sovity-daps`, but rather already has a running instance of Keycloak 26 and wants the DAT Mapper available there, under "Configure a new mapper". It can be assumed that a "DAPS" realm will be created in this existing Keycloak instance and that everything will be handled within the "DAPS" realm according to the guidelines.

**Comment by ghost** [comment, marked as answer]
Hello, as outcome: the DAT Mapper is an extension to a Keycloak and the source code can be found here, so we recommend using "our" approach instead of going the manual route: https://github.com/sovity/sovity-daps/blob/main/src/main/java/de/sovity/keycloak/mapper/dat/DatMapper.java

**Reply by xnakos** [reply]
Thank you, Tim. After more thought, I came to the conclusion that the DAPS Keycloak and the DAPS realm should be as standalone and isolated as possible, so the integration into an existing Keycloak should not be pursued.
